collect_app/src/main/java/org/odk/collect/android/widgets/ExStringWidget.java

```diff
@@ -265,9 +265,15 @@ public void onButtonClick(int buttonId) {
             }
         } else {
             onException(errorString);
+            showKeyboard(getContext());
```

Comment:
I think it should be in `onException()` since we call  `answer.requestFocus()` there too.